### PR TITLE
Use licence MPL 2.0 instead of GPL3_WITH_RTE-Exceptions

### PR DESCRIPTION
**Pull request by flomnes**
As of [YYYY/MM/DD], RTE Group, holding exclusive copyright rights, has announced a significant update in their software licensing policy for Antares Simulator. With the release of the major 9.0 version of Antares Simulator, RTE Group is transitioning from the GNU General Public License Version 3 (GPL-3) to the Mozilla Public License Version 2.0 (MPL-2.0). This strategic move reflects RTE Group's ongoing commitment to fostering an open, collaborative development environment while ensuring a balanced approach to copyright management and distribution flexibility.

### CODE REVIEW

**Comment by flomnes** [comment]
We need a major version for this, so that would be 9.0

**Comment by a-zakir** [comment]
> This has been the largest search&replace in the history of search&replaces, maybe ever !

congrats m8

**Review by a-zakir — APPROVED** [review]
definitely agree!

**Review by 0x62646f727465 — CHANGES_REQUESTED** [review]
My review as the OSPO Director at RTE:

- [x] 0) Not only src subdirectroy, please consider the docs subdirectory and root md files also

Example for doc subdir: docs/reference-guide/12-attribution_notices.md presents GNU GPL informations.
Example for /*.md file: README.md 

- [x] 1) Correct the Copyright notice

Please respect this exact pattern:

Copyright (c) [FIRSTYEAR]-2023, RTE (https://www.rte-france.com)
See AUTHORS.txt
SPDX-License-Identifier: MPL-2.0
This file is part of Antares-Simulator, Adequacy and Performance assesment for interconnected energy networks.

[FIRSTYEAR] must be the oldest year of the copyright to consider in the updated file
As we are releasing a major version we will update the Copyright year
If you will release the 9.0 version after January 1st then it will be 2024
Warning: do not update the copyright of others (ex. Matsumoto's once)

- [x] 2) Correct the SPDX-License-Identifier from MPL<space>2.0 to MPL<dash>2.0

Please replace all "SPDX-License-Identifier: MPL 2.0" by "SPDX-License-Identifier: MPL-2.0"

- [x] 3) Post commit please correct the published documentation

- [x] 3.a) Readthedocs.io

Correct Flags from GPLv3 to MPLv2
Replace GPL V3 by MPL V2

- [x] at https://antares-simulator.readthedocs.io/en/latest/
After the line "In May 2018 RTE decided to release the project under the GPLv3 license."
Add the line "In [DATE], RTE, as the exclusive Copyright owner decided, to switch from GPL V3 to MPL V2 License since the 9.0 major version of Antares-Simulator"

- [ ] 3.b) https://antares-simulator.org/

- [x] 4) Exclude src/ext/yuni

No changes should be made in license.md

- [x] 5) Double check src/libs/antares/*

For example, Merseen-Twister appears to be a BSD Licence and there's no reference to it

- [x] 6) commit and commit comment

The Version 9.0 changelog should be dedicated to the License change
No feature should be mixed with. A 9.1 version will.

The comment of this commit should precise:
As of [YYYY/MM/DD], RTE Group, holding exclusive copyright rights, has announced a significant update in their software licensing policy for Antares-Simulator. With the release of the major 9.0 version of Antares-Simulator, RTE Group is transitioning from the GNU General Public License Version 3 (GPLv3) to the Mozilla Public License Version 2.0 (MPL V2.0). This strategic move reflects RTE Group's ongoing commitment to fostering an open, collaborative development environment while ensuring a balanced approach to copyright management and distribution flexibility.

**Comment by JasonMarechal25** [comment]
What about files without licence head ?

File: src/libs/antares/benchmarking/DurationCollector.cpp [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
```
Comment (flomnes) [inline src/libs/antares/benchmarking/DurationCollector.cpp]: Remove

File: .gitignore [hunk 2]
```diff
@@ -55,6 +55,7 @@ src/script/v8/libs/*
 /src/antares.VC.db
 
 _build*
+tags
```
Comment (flomnes) [inline .gitignore]: Revert

**Pull request merged**